includes/class-wc-payments-token-service.php

```diff
@@ -71,17 +71,15 @@ public function add_token_to_user( $payment_method, $user ) {
 		switch ( $payment_method['type'] ) {
 			case Payment_Method::SEPA:
 				$token      = new WC_Payment_Token_WCPay_SEPA();
-				$gateway_id = WC_Payments_Features::is_upe_split_enabled() ?
+				$gateway_id = WC_Payments_Features::is_upe_split_enabled() || WC_Payments_Features::is_upe_deferred_intent_enabled() ?
 					WC_Payment_Gateway_WCPay::GATEWAY_ID . '_' . Payment_Method::SEPA :
 					CC_Payment_Gateway::GATEWAY_ID;
 				$token->set_gateway_id( $gateway_id );
 				$token->set_last4( $payment_method[ Payment_Method::SEPA ]['last4'] );
 				break;
 			case Payment_Method::LINK:
 				$token      = new WC_Payment_Token_WCPay_Link();
-				$gateway_id = WC_Payments_Features::is_upe_split_enabled() ?
-					WC_Payment_Gateway_WCPay::GATEWAY_ID . '_' . Payment_Method::LINK :
-					CC_Payment_Gateway::GATEWAY_ID;
```

Comment:
Does this change have any repercussions on Stripe Link tokens that have already been saved?

Reply:
Good question!

For payment tokens at checkout, these will automatically be replaced, since whenever we list out saved payment tokens, we collect a list from Stripe, compare this list to what we have stored, and [create new tokens for anything that is missing](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-token-service.php#L197). This means that we create new valid tokens to take the place of the original invalid tokens. 

You can confirm by this by adding a saved payment method before Stripe on `develop`, where it will be hidden: upon switching to this branch it will magically reappear.

But what does this mean for existing saved payment tokens used with subscriptions? I tested this out by paying for a subscription product using Link and then switching to this branch and trying to renew the subscription and a few peculiar things happened. 

As I mentioned in [this comment](https://github.com/Automattic/woocommerce-payments/issues/6261#issuecomment-1682899459), the renewal _did_ actually process successfully. A new renewal order was created with accompanying payment intent--all of which were successful. I think this might be because the payment token still has a valid, reusable payment method attached to it, so WC Subscriptions somehow manages to make it work. 

However, there was an error notice at the top of the subscription order after I processed the renewal successfully, stating that the payment method on the subscription was missing. Following the prompt, I was this time able to update the saved payment method on the subscription to the proper saved Stripe token using the dropdown in the billing information, since this token now appears under the correct gateway. 

![Updating subscription payment method dropdown](https://cdn-std.droplr.net/files/acc_1104206/po7I4W)

Compare the image above to the second screenshot in Chris's original issue description over [here](https://github.com/Automattic/woocommerce-payments/issues/6261#issue-1702849296), where the store owner is unable to update the saved payment method to the desired method.

TL;DR: subscription renewals appear to process successfully with a non-breaking error that prompts store owner to manually fix the subscription, thanks to the fix in this PR. It's not really working by intention...but is working without breaking and suggests an intuitive workaround to permanently resolve itself. 🤷 

If this is deemed acceptable, maybe we don't need to worry about manually migrating payment tokens--SEPA tokens included. 🤔

Reply:
Thank you for the investigation and explanation! The workaround following the error message sounds acceptable as well! 👍